lib/internal/modules/cjs/loader.js

```diff
@@ -297,8 +327,58 @@ function findLongestRegisteredExtension(filename) {
   return '.js';
 }
 
+// This only applies to requests of a specific form:
+// 1. name/.*
+// 2. @scope/name/.*
+const EXPORTS_PATTERN = /^((?:@[^./@\\][^/@\\]*\/)?[^@./\\][^/\\]*)(\/.*)$/;
+function resolveExports(nmPath, request, absoluteRequest) {
+  // The implementation's behavior is meant to mirror resolution in ESM.
+  if (experimentalExports && !absoluteRequest) {
+    const [, name, expansion] =
+      request.match(EXPORTS_PATTERN) || [];
+    if (!name) {
+      return path.resolve(nmPath, request);
+    }
+
+    const basePath = path.resolve(nmPath, name);
+    const pkgExports = readExports(basePath);
+
+    if (pkgExports != null) {
+      const mappingKey = `.${expansion}`;
+      const mapping = pkgExports[mappingKey];
+      if (typeof mapping === 'string') {
+        return path.resolve(basePath, mapping);
+      }
+
+      let dirMatch = '';
+      for (const [candidateKey, candidateValue] of Object.entries(pkgExports)) {
```

Comment:
Grab `Object.entries` off primordials to avoid mutation / interception

Reply:
`Object` is pulled from `primordials` above - I assume that's all that is needed?

Reply:
oh, must have missed that